### ISSUE TITLE
Suicide Squad Isekai is pulling the Japanese episode summaries from TVDB

### ISSUE THREAD

**Issue by Jinra** [ZeroQI/Hama.bundle#576]
Oddly enough, the series summary is in english, but episode summaries are pulling from the Japanese version. I've done quite a bit of testing in the bundle, but can't figure out why it only wants to pull Japanese.

I've tested on another anime that's inheriting from TVDB (Berserk 1997), but that pulls both series and episode summaries in English as expected. Any API calls either from CLI or browser, both pull summaries in english, even when copying the exact headers and token the bundle is using.

Here's a python snippet I've been using for testing. The only way I can get it to pull in Japanese is to change the `Accept-Language` key value to `ja`

```python
import json
import requests

url = 'https://api.thetvdb.com'
test_ep = '/series/436846/episodes?page=1'
token = 'A27AD9BE0DA63333'
login_ep = '/login'
headers = {
    'User-agent': 'Plex/HAMA',
    'Content-type': 'application/json',
    'Accept-Language': 'en'
}
bearer_token = ''

data = {
    'apikey': token
}

if not bearer_token:
    r = requests.post(url+login_ep, headers=headers, json=data)
    bearer_token = r.json()['token']

headers['Authorization'] = f'Bearer {bearer_token}'
r = requests.get(url+test_ep, headers=headers)
pass```

**Comment by ZeroQI** [comment]
Plex library language is the one used for the summaries, whether series or episodes.
Metadata source for each is not indicated.

> I've done quite a bit of testing in the bundle, but can't figure out why it only wants to pull Japanese.

If you did look in the series specific logs, it tells the metadata source priority for each field AND which source provided which value... Logs location is in the troubleshooting section of the readme

**Comment by Jinra** [comment]
> If you did look in the series specific logs, it tells the metadata source priority for each field AND which source provided which value

Yes and it is as I mentioned, they're both pulling from TVDB. I'm just not sure why it's pulling the Japanese version

> Plex library language is the one used for the summaries, whether series or episodes.

Library is set to English, and language priorities are only `en`

**Comment by ZeroQI** [comment]
https://api.thetvdb.com/series/436846/episodes?page=1 doesn't have japanese summary, title pulled from AniDB.

Cached file japanese indeed
TheTVDB/json/436846/episode_9879316_ja.json

Don't have logs so cannot check the json for each episode by clicking on it to identify the issue...

**Comment by Jinra** [comment]
> https://api.thetvdb.com/series/436846/episodes?page=1 doesn't have japanese summary, title pulled from AniDB.

Yea I'm only talking about the summaries. There is a Japanese summary which is what Hama is pulling for `episodes_page1-en.json` attached here [episodes_page1_en.json](https://github.com/user-attachments/files/16716663/episodes_page1_en.json)

> Don't have logs so cannot check the json for each episode by clicking on it to identify the issue...

Here are the full logs
[Suicide Squad ISEKAI {imdb-tt28279848}.agent-update.log](https://github.com/user-attachments/files/16716680/Suicide.Squad.ISEKAI.imdb-tt28279848.agent-update.log)

**Comment by Jinra** [comment]
After some more digging and testing. I think the problem is the line here https://github.com/ZeroQI/Hama.bundle/blob/daa43001bc1ced67aa2f90de8c61e5a1d109e862/Contents/Code/TheTVDBv2.py#L234

`lang2 = 'en' if len(language_episodes)<=1 else language_episodes[1]`

This seems to determine the search language. I have my priority setup as `en, x-jat, ja`. Since line 71 defining `language_episodes` seems to ignore x-jat, the remaining array is just `['en', 'ja']`. The logic in line 234 seems to ignore the first priority item entirely and use the language at index 1 which is `ja` for me.

**Comment by ZeroQI** [comment]
common.SaveFile() - CachePath: '/config/Library/Application Support/Plex Media Server/Plug-in Support/Data/com.plexapp.agents.hama/DataItems', file: 'TheTVDB/json/436846/episodes_page1_en.json'
=> the file says english but contain Japanese summaries, is loaded at that line, and use lang variable (the library language which is set to english)

https://github.com/ZeroQI/Hama.bundle/blob/daa43001bc1ced67aa2f90de8c61e5a1d109e862/Contents/Code/TheTVDBv2.py#L129

delete the cached file, metadata search for series and modify title searched to "clear-cache", cancel, and refresh metadata and check this json file content for english summary

**Comment by ZeroQI** [comment]
Issue still present ?

**Comment by Jinra** [comment]
it might be, but I've adjusted my language to only have `en` as a workaround
